### ISSUE TITLE
Typescript Service died X times...

### ISSUE THREAD

**Issue by izoran** [Microsoft/vscode#40849]
- VSCode Version: Code 1.19.1 (0759f77bb8d86658bc935a10a64f6182c5a1eeba, 2017-12-19T09:35:14.671Z)
- OS Version: Darwin x64 15.6.0
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
EditorConfig|Edi|0.11.1
sdlang|Lau|0.1.2
haskelly|UCL|0.4.29
gitignore|cod|0.5.0
scala|dal|0.0.4
dlang|dla|0.9.0
code-runner|for|0.8.6
sml|fre|0.0.18
language-haskell|jus|2.4.0
nim|kos|0.5.27
vscode-elixir|mjm|1.1.0
python|ms-|0.9.1
vshaxe|nad|1.9.3
erlang|pgo|0.1.3
Ruby|reb|0.15.0
elm|sbr|0.15.1
slim|sia|0.1.2

---

**Comment by mjbvz** [comment]
A few questions:

- Does this happen if you disable all extensions?
- Does this happen for all projects you open, or just one specifically?
- Have you set a custom `typescript.tsdk` value in your workspace or user settings?

Can you also please try collecting the TypeScript server logs:

1. Add `"typescript.tsserver.log": "verbose"` to your user settings and restart VSCode.
2. Reproduce the issue
3. Back in the `TypeScript` section of the output panel, look for a `TSServer exited` message. Right after that message there should now be the path to a log printed. Open that log directory and look in the `tsserver.log` file

Look for errors, stack traces, or other suspicious info near the end of the logs. If you can share the logs I can try looking through them to see what is going on

 > **⚠️Warning:** The TypeScript log may include information from your workspace, including file paths and source code. If you have any concerns about posting this publicly on Github, just let me know and we can arrange something else. On our side, we only use these logs to investigate issues like this

**Comment by izoran** [comment]
Hi there,

first off, the issue kept happening sporadically but I couldn't really say when or why.

And when I got the prompt to report the issue for Nth time, I just clicked "yes" thinking it would simply submitted the usual automatically gathered exception information like I was used to from other Microsoft applications. Instead I landed on GitHub, in an issue report editor. I could give you more details at that time than what VSCode prefilled though.

Meanwhile I figured out it only started after I installed the [Haselly] extension], and it stopped when I disabled it.

As long as Haskelly was enabled, it kept happening on all types of files, not only Haskell files.

[Haskelly]: https://marketplace.visualstudio.com/items?itemName=UCL.haskelly

It also happened in editor sessions without ever opening any Haskell files. I know I restarted VSCode after the exception, and only opened a non-Haskell file. I still got the exception. In retrospect it seems weird b/c I'd expect extensions are loaded on demand. The directory tree did contain Haskell files though.

That's all I can say about the issue. I hope it helps.
